lib/CMakeLists.txt

```diff
@@ -231,7 +231,7 @@ target_include_directories(quda PRIVATE .)
 target_include_directories(quda PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
 target_include_directories(quda SYSTEM PRIVATE ../include/externals)
 target_include_directories(quda SYSTEM PUBLIC $<$<COMPILE_LANGUAGE:CUDA>:${CUDAToolkit_INCLUDE_DIRS}>)
-target_link_libraries(quda PRIVATE Eigen)
+target_include_directories(quda SYSTEM PRIVATE ${EIGEN_INCLUDE_DIRS})
```

Comment:
@cpviolator I'd like to revert this change as a private dependency should not be propagated to a downstream build. If you have a reproducer for the issue can you share the details?

Reply:
I've attached a reproducer.
[compile_eigen_repro.txt](https://github.com/lattice/quda/files/7471240/compile_eigen_repro.txt)
The script assumes:
```
Currently Loaded Modules:
  1) StdEnv (S)   2) cmake/3.20.2   3) cuda/11.4.1   4) gcc/8.3.1   5) spectrum-mpi/rolling-release
```
Everything is automatic except at the point where the revert is applied, then you have to interact to continue. The error I see is:
```
[100%] Linking CXX executable t_minvert
/usr/tce/packages/gcc/gcc-8.3.1/rh/usr/bin/../libexec/gcc/ppc64le-redhat-linux/8/ld: cannot find -lEigen
collect2: error: ld returned 1 exit status
make[2]: *** [mainprogs/tests/CMakeFiles/t_minvert.dir/build.make:116: mainprogs/tests/t_minvert] Error 1
make[1]: *** [CMakeFiles/Makefile2:782: mainprogs/tests/CMakeFiles/t_minvert.dir/all] Error 2
```
for each of the chroma executables.

Reply:
Thanks. Fixed in 6736491